contracts/sentinel/guards/FuseWithdrawalGuard.sol

```diff
@@ -0,0 +1,131 @@
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    struct WithdrawInfo {
+        address destination;
+        address underlying;
+        uint96 liquidityToLeave;
+    }
+
+    /// @notice map the destination and minimum liquidity for each pcv deposit
+    mapping(address => WithdrawInfo) public withdrawInfos;
+
+    EnumerableSet.AddressSet private fuseDeposits;
+
+    /// @notice the PCV mover contract exposed to guardian role
+    PCVGuardian public constant pcvGuardian = PCVGuardian(0x02435948F84d7465FB71dE45ABa6098Fc6eC2993);
+
+    constructor(
+        address core,
+        address[] memory deposits,
+        address[] memory destinations,
+        address[] memory underlyings,
+        uint96[] memory liquidityToLeaveList
+    ) CoreRef(core) {
+        uint256 len = deposits.length;
+        require(len == destinations.length && len == liquidityToLeaveList.length && len == underlyings.length);
+        for (uint256 i = 0; i < len; ) {
+            fuseDeposits.add(deposits[i]);
+            withdrawInfos[deposits[i]] = WithdrawInfo({
+                destination: destinations[i],
+                underlying: underlyings[i],
+                liquidityToLeave: liquidityToLeaveList[i]
+            });
+            unchecked {
+                ++i;
+            }
+        }
+    }
+
+    /// @notice setter for the Fuse deposit destination and minimum liquidity
+    function setWithdrawInfo(address deposit, WithdrawInfo calldata withdrawInfo)
+        public
+        onlyTribeRole(TribeRoles.GUARDIAN)
+    {
+        withdrawInfos[deposit] = withdrawInfo;
+    }
+
+    /// @notice helper method to trustlessly and publicly remove deposits once emptied.
+    /// TODO could be removed to simplify/minimize attack surface and public functions.
+    function removeDepositIfEmpty(ERC20CompoundPCVDeposit deposit) public {
+        if (deposit.balance() == 0) {
+            delete withdrawInfos[address(deposit)];
+        }
+    }
+
+    /// @notice check if contract can be called. If any deposit has a nonzero withdraw amount available, then return true.
+    function check() external view override returns (bool) {
+        for (uint256 i = 0; i < fuseDeposits.length(); ) {
+            if (getAmountToWithdraw(ERC20CompoundPCVDeposit(fuseDeposits.at(i))) > 0) return true;
+            unchecked {
+                ++i;
+            }
+        }
+        return false;
+    }
+
+    /// @notice return the amount that can be withdrawn from a deposit after leaving min liquidity
+    function getAmountToWithdraw(ERC20CompoundPCVDeposit deposit) public view returns (uint256) {
+        // can't read underlying directly because some of the PCV deposits use an old abi
+        IERC20 underlying = IERC20(withdrawInfos[address(deposit)].underlying);
+        // Reserves of underlying left in the cToken are considered withdrawable liquidity
+        uint256 liquidity = underlying.balanceOf(address(deposit.cToken()));
```

Comment:
is that correct ? ctokens should be holding the platform fees & reserves too ?

Reply:
They are withdrawable either way. Platform fees do not have a liquidity preference